### PR TITLE
Misc bugfix

### PR DESCRIPTION
**Pull request by jdef**
Fix some core bugs that I stumbled into while working on dnstap stuff.

### CODE REVIEW

File: resolver/resolver.go [hunk 1]
```diff
@@ -360,15 +360,15 @@ func (res *Resolver) HandleMesos(w dns.ResponseWriter, r *dns.Msg) {
 	name := strings.ToLower(cleanWild(r.Question[0].Name))
 	switch r.Question[0].Qtype {
 	case dns.TypeSRV:
-		errs = append(errs, res.handleSRV(rs, name, m, r))
+		errs.Add(res.handleSRV(rs, name, m, r))
```
Comment (tsenart) [inline resolver/resolver.go]: What was the bug you found here?

File: resolver/resolver.go [hunk 2]
```diff
@@ -710,6 +710,22 @@ func cleanWild(name string) string {
...
+func (e multiError) Add(err error) multiError {
+	if me, ok := err.(multiError); ok {
+		return append(e, me...)
+	} else if err != nil {
+		return append(e, err)
+	}
+	return e
+}
+
+func (e multiError) AddAll(err ...error) multiError {
```
Comment (tsenart) [inline resolver/resolver.go]: There's no need for having both `AddAll` and `Add`.

File: records/config.go [hunk 3]
```diff
@@ -159,12 +159,15 @@ func readConfig(file string) (*Config, error) {
 		RecurseOn:      true,
 	}
 
+	workingDir := "."
 	usr, err := user.Current()
 	if err != nil {
-		return nil, fmt.Errorf("couldn't retrieve current user: %v", err)
+		logging.Error.Println("Failed to determine current user, translating ~/ to ./, error was", err)
```
Comment (tsenart) [inline records/config.go]: Why was this a bug before?

Comment (jdef) [inline records/config.go]: if I start mesos-dns as non-root user then Current() returns nil for me

File: resolver/resolver.go [hunk 4]
```diff
@@ -377,14 +377,14 @@ func (res *Resolver) HandleMesos(w dns.ResponseWriter, r *dns.Msg) {
...
-		errs = append(errs, res.handleEmpty(rs, name, m, r))
+		errs.Add(res.handleEmpty(rs, name, m, r))
 	} else {
 		shuffleAnswers(res.rng, m.Answer)
 		logging.CurLog.MesosSuccess.Inc()
 	}
 
 	if !errs.Nil() {
-		logging.Error.Println(errs)
+		logging.Error.Println(errs.Error())
```
Comment (jdef) [inline resolver/resolver.go]: the symptom of the bug was showing here: blank error lines

File: resolver/resolver.go (hunk 2 above)
Comment (jdef) [inline resolver/resolver.go]: agreed, will consolidate

Comment (jdef) [inline resolver/resolver.go]: and i'll add a test case for the bug that i'm fixing with this impl

Comment (tsenart) [inline resolver/resolver.go]: :clap:

File: resolver/resolver.go (hunk 4 above)
Comment (jdef) [inline resolver/resolver.go]:
```
slave1_1      | ERROR: 2015/08/12 19:55:55 resolver.go:387: ; 
slave1_1      | ERROR: 2015/08/12 19:55:55 resolver.go:387: ; 
slave1_1      | ERROR: 2015/08/12 19:55:55 resolver.go:387: ; 
```

**Pull request merged**